finality-aleph/src/metrics/chain_state.rs

```diff
@@ -135,59 +186,57 @@ pub async fn run_chain_state_metrics<
     TP: TransactionPoolInfoProvider<Extrinsic = X>,
 >(
     backend: &BE,
-    import_notifications: ImportNotifications<B>,
-    finality_notifications: FinalityNotifications<B>,
+    mut import_notifications: ImportNotifications<B>,
+    mut finality_notifications: FinalityNotifications<B>,
     registry: Option<Registry>,
     mut transaction_pool_info_provider: TP,
-) {
+) -> Result<(), ChainStateMetricsError> {
+    use ChainStateMetricsError as Error;
+
     if registry.is_none() {
-        return;
+        return Err(Error::no_registry());
+    }
+    if import_notifications.is_terminated() {
+        return Err(Error::block_import_stream_closed());
+    }
+    if finality_notifications.is_terminated() {
+        return Err(Error::finalized_blocks_stream_closed());
     }
 
     let metrics = match ChainStateMetrics::new(registry) {
         Ok(metrics) => metrics,
         Err(e) => {
             warn!(target: LOG_TARGET, "Failed to create metrics: {e}.");
-            return;
+            return Err(Error::unable_to_create_metrics(e));
```

Comment:
You can probably `map_err()?` here, instead of this whole `match`. The log ain't that useful and should be duplicated by error handling outside anyway.

Reply:
done